flaml/automl/automl.py

```diff
@@ -3557,6 +3569,13 @@ def _search_sequential(self):
                     state.best_config,
                     self.data_size_full,
                 )
+
+                if (
+                    self._trained_estimator.params[self._trained_estimator.ITER_HP]
+                    != self.best_config[self._trained_estimator.ITER_HP]
```

Comment:
These are not safe. Multiple errors could happen. `self._trained_estimator` may not have `params` or `ITEM_HP`.
`self._trained_estimator.params` or `self.best_config` may not contain `self._trained_estimator.ITER_HP`.